### PR TITLE
Only set oidc flags when the OpenIDAuth feature is enabled

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

Because of this the kubermatic controller manager on cloud asia and cloud us is currently crashlooping. On cloud euw we fortunately hit some issue with GKE with make port-forwarding not work, hence helm didn't do anything there. Please just merge this after approving it.

**Documentation**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by p0lyn0mial — APPROVED** [review]
so sorry for the trouble

**Comment by p0lyn0mial** [comment]
btw is there a way to test if the chart will be valid (syntax) locally ?

**Comment by p0lyn0mial** [comment]
@alvaroaleman thanks for fixing this !

**Comment by alvaroaleman** [comment]
> btw is there a way to test if the chart will be valid (syntax) locally ?

Sort of, I tested this by executing a `helm template . |grep oidc-issuer` in the `config/kubermatic` directory and veryfying I got nothing, then doing a `helm template . --set kubermatic.controller.featureGates="OpenIDAuthPlugin=true"|grep oidc-issuer` and veryfing I got something

File: config/kubermatic/templates/kubermatic-api-dep.yaml [hunk 1]
```diff
@@ -36,10 +36,12 @@ spec:
           - -master-resources=/opt/master-files
           # the following flags enable oidc kubeconfig feature/endpoint
           - -feature-gates={{ .Values.kubermatic.api.featureGates }}
+          {{- if regexMatch ".*OpenIDAuthPlugin=true.*" (default "" .Values.kubermatic.controller.featureGates) }}
```
Comment (p0lyn0mial) [inline config/kubermatic/templates/kubermatic-api-dep.yaml:39]: for api server this flag has a different name  `OIDCKubeCfgEndpoint`
https://github.com/kubermatic/kubermatic/blob/master/api/cmd/kubermatic-api/features.go#L10

Comment (alvaroaleman) [reply config/kubermatic/templates/kubermatic-api-dep.yaml:39]: Oh sorry, I already merged it :( Will fix it in a follow-up

**Pull request merged**